plugin/cmd/kube-scheduler/app/server.go

```diff
@@ -76,17 +77,20 @@ func Run(s *options.SchedulerServer) error {
 	// cache only non-terminal pods
 	podInformer := factory.NewPodInformer(kubecli, 0)
 
+	var configMapInformer cache.SharedIndexInformer
+	if !s.UseLegacyPolicyConfig && len(s.PolicyConfigMapName) != 0 {
+		// cache only scheduler's policy ConfigMap
+		configMapInformer = factory.NewPolicyConfigMapInformer(kubecli, 0, s.PolicyConfigMapName, s.PolicyConfigMapNamespace)
+	} else {
+		configMapInformer = nil
+	}
+
 	sched, err := CreateScheduler(
 		s,
 		kubecli,
-		informerFactory.Core().V1().Nodes(),
+		informerFactory,
 		podInformer,
-		informerFactory.Core().V1().PersistentVolumes(),
-		informerFactory.Core().V1().PersistentVolumeClaims(),
-		informerFactory.Core().V1().ReplicationControllers(),
-		informerFactory.Extensions().V1beta1().ReplicaSets(),
-		informerFactory.Apps().V1beta1().StatefulSets(),
-		informerFactory.Core().V1().Services(),
+		configMapInformer,
```

Comment:
I wouldn't expect the configmap watcher to leak into the scheduler here... the scheduler shouldn't care about the source of its config struct